third_party/move/move-compiler-v2/src/pipeline/split_critical_edges_processor.rs

```diff
@@ -0,0 +1,193 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! This pass splits critical edges with empty blocks.
```

Comment:
Done. It may be better for future transformers to be able to decide if they want to retain the no-critical-edges property, so I just doced here that after the transformation there is no critical edges.